File: clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (let 
    [modifier (cond 
                (< speed 1) 0
                (<= speed 4) 1
                (<= speed 8) 0.9
                (= speed 9) 0.8
                :else 0.77)]
    (* speed 221 modifier))
  )

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (/ (production-rate speed) 60))
  )

File: go/cars-assemble/cars_assemble.go
package cars

// CalculateWorkingCarsPerHour calculates how many working cars are
// produced by the assembly line every hour.
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * (successRate / 100.0);
}

// CalculateWorkingCarsPerMinute calculates how many working cars are
// produced by the assembly line every minute.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(CalculateWorkingCarsPerHour(productionRate, successRate) / 60.0)
}

// CalculateCost works out the cost of producing the given number of cars.
func CalculateCost(carsCount int) uint {
	var res int

    if int(carsCount / 10) > 0 {
        res += int(carsCount / 10) * 95000
        carsCount -= carsCount - (carsCount % 10)
    }

    if carsCount > 0 {
        res += carsCount * 10000
    }

    return uint(res)
}
